client/packages/programs/src/VaccineCourseEditModal/VaccineCourseEditModal.tsx

```diff
@@ -217,3 +216,125 @@ export const VaccineCourseEditModal: FC<VaccineCourseEditModalProps> = ({
     </Modal>
   );
 };
+
+const VaccineCourseDoseTable = ({
+  doses,
+  updatePatch,
+  courseName,
+}: {
+  courseName: string;
+  doses: VaccineCourseScheduleFragment[];
+  updatePatch: (newData: Partial<DraftVaccineCourse>) => void;
+}) => {
+  const t = useTranslation('programs');
+
+  const addDose = () => {
+    updatePatch({
+      vaccineCourseSchedules: [
+        ...doses,
+        {
+          __typename: 'VaccineCourseScheduleNode',
+          id: FnUtils.generateUUID(),
+          // temp - will be overwritten by the backend to assign unique dose number (even if previous doses were deleted)
+          doseNumber: doses.length + 1,
+          label: `${courseName} ${doses.length + 1}`,
+          minAgeMonths: 0,
+          minIntervalDays: 0,
+        },
+      ],
+    });
+  };
+
+  const deleteDose = (id: string) => {
+    updatePatch({
+      vaccineCourseSchedules: doses.filter(dose => dose.id !== id),
+    });
+  };
+
+  const updateDose = (
+    id: string,
+    newData: Partial<VaccineCourseScheduleNode>
+  ) => {
+    updatePatch({
+      vaccineCourseSchedules: doses.map(dose =>
+        dose.id === id ? { ...dose, ...newData } : dose
+      ),
+    });
+  };
+
+  return (
+    <>
+      <Box display={'flex'} justifyContent={'flex-end'}>
+        <ButtonWithIcon
+          Icon={<PlusCircleIcon />}
+          label={t('label.dose')}
+          onClick={addDose}
+        />
+      </Box>
+      <Table sx={{ marginTop: '5px', '& td': { padding: '3px' } }}>
+        <tr style={{ borderBottom: '1px solid lightgray' }}>
+          <HeaderCell label="Dose #" width="80px" />
+          <HeaderCell label={t('label.label')} width="250px" />
+          <HeaderCell label="Age (months)" />
+          <HeaderCell label="Min interval (days)" />
+        </tr>
+        <tbody>
+          {doses.map((dose, index) => (
+            <tr>
+              <td style={{ textAlign: 'center' }}>{index + 1}</td>
+              <td>
+                <BasicTextInput
+                  value={dose.label}
+                  fullWidth
+                  onChange={e => updateDose(dose.id, { label: e.target.value })}
+                />
+              </td>
+              <NumericDoseTableCell
+                value={dose.minAgeMonths}
+                onChange={num => updateDose(dose.id, { minAgeMonths: num })}
+              />
+              <NumericDoseTableCell
+                value={dose.minIntervalDays}
+                onChange={num => updateDose(dose.id, { minIntervalDays: num })}
+              />
+              <td style={{ display: 'flex', justifyContent: 'center' }}>
+                <IconButton
```

Comment:
Is there anywhere else we've used the Delete icon like this? (I can't find one). If so, I guess just copy that, but if not, I feel like it could be a tiny bit smaller? Not totally sure though.